inst/schema/Graph.schema.json

```diff
@@ -46,11 +46,52 @@
             "type": "string"
           }
         },
-        "id": {"type:" :"string"},
+        "id": {"type":"string"},
         "name": {"type":  "string"},
         "type": {"type": "string"}
       },
       "additionalProperties": true
+    },
+    "layout": {
+      "type": "object",
+      "properties": {
+        "xaxis": {
+          "$ref": "#/definitions/axis"
+        },
+        "yaxis": {
+          "$ref": "#/definitions/axis"
+        },
+        "shapes" : {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "type": {"type": "string"},
+              "y_formula": {"type": "string"},
+              "x0": {"type": "number"},
+              "x1": {"type": "number"},
+              "y0": {"type": "number"},
+              "y1": {"type": "number"}
+            },
+            "additionalProperties": true
+          }    
+        }  
+      },
+      "additionalProperties": true
+    },
+    "axis": {
+      "type": "object",
+      "properties": {
+        "autorange": {"type": "boolean"},
+        "rangemode": {"type": "string"},
+        "range": {
+          "type": "array",
+          "items": {
+            "type": "number"
+          }
+        }
+      },
+      "additionalProperties": true
```

Comment:
Are any of these fields mandatory? If so then a `required` array might be helpful?

Reply:
Yeah we're very tolerant atm, more so than we need to be probably. I guess there's some basic fields that sensibly need to be there. Will add for these new types, should probably audit for the existing schema.